fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -1344,6 +1545,31 @@ public void dropPartition(long dbId, String partitionName, boolean isForceDrop)
         dropPartition(dbId, partitionName, isForceDrop, false);
     }
 
+    /**
+     *  Try to retain the partition for a while to avoid tablet missing errors.
+     *  This method is used in `insert overwrite`, `mv rewrite` etc. scenarios,
+     *  where we don't want to delete data immediately
+     */
+    public void dropPartitionGracefully(long dbId, String partitionName, long partitionRetentionPeriod) {
+        Partition partition = nameToPartition.get(partitionName);
```

Comment:
will this be performed under lock? or the caller will be properly handle the lock? Not quite sure.